### DISCUSSION TITLE
Site Optimization

### DISCUSSION THREAD

**Discussion by viragjain** [vercel/next.js#16172] [Help]
I am using Google tag manager and some other tracking tools, because of that site performance is decreased very much.

I am currently loading the tracking scripts in componentDidMount() under pages/index.js, please suggest the optimal way of loading the GTM scripts.

Thanks in advance!!

**Comment by jamesmosier** [comment]
Take a look at the [with-google-analytics](https://github.com/vercel/next.js/tree/canary/examples/with-google-analytics) example. This shows how to setup up the script imports to be async in [_document.js](https://github.com/vercel/next.js/blob/canary/examples/with-google-analytics/pages/_document.js)

**Reply by viragjain** [reply]
Thanks @jamesmosier I will try loading this in _document.
Could you please take a look on gps report? We have lazyImageLoad, gzip compression, webp image setup and all present, but performance is still very much low.

https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Fwwwinvisalign-conte-test-eu.herokuapp.com%2F
